### PR TITLE
[IMPROVE] Add support to queries in `channels.members` and `groups.members` endpoints

### PR DESCRIPTION
**Pull request by matheusbsilva137**
## Proposed changes (including videos or screenshots)

 - Add support to queries (within the `query` parameter) in `channels.members` and `groups.members` endpoints.

## Issue(s)

[Task - ClickUp](https://app.clickup.com/t/3z2ttu)

## Steps to test or reproduce

Example input (using cURL):
```
curl -G 'http://localhost:3000/api/v1/channels.members?roomId=vrdXhpY2BvB8R8MMF&status[]=offline&username=some.user' \
  -H 'X-Auth-Token: Dommgtm9xJefbcn7vyjVi2PbrM_Tk3SAnYl043qkflp' \
  -H 'X-User-Id: dizcH9Edm8hC7ziDX' \
  -H 'Content-Type: application/json'
```

### CODE REVIEW

File: app/api/server/v1/channels.js [hunk 1]
```diff
@@ -588,8 +593,13 @@ API.v1.addRoute('channels.members', { authRequired: true }, {
 		const total = subscriptions.count();
 
 		const members = subscriptions.fetch().map((s) => s.u && s.u._id);
+		const query = {
+			name,
```
Comment (KevLehman) [inline app/api/server/v1/channels.js:597]: Is expected that `name` and `username` are not regexp?

File: app/api/server/v1/channels.js [hunk 2]
```diff
@@ -577,6 +577,11 @@ API.v1.addRoute('channels.members', { authRequired: true }, {
 
 		const { offset, count } = this.getPaginationItems();
 		const { sort = {} } = this.parseJsonQuery();
+		const { status, username, name } = this.queryParams;
+
+		if (status && !Array.isArray(status)) {
```
Comment (KevLehman) [inline app/api/server/v1/channels.js:582]: Use `check` here.

File: app/api/server/v1/groups.js [hunk 3]
```diff
@@ -500,6 +500,11 @@ API.v1.addRoute('groups.members', { authRequired: true }, {
 
 		const { offset, count } = this.getPaginationItems();
 		const { sort = {} } = this.parseJsonQuery();
+		const { status, username, name } = this.queryParams;
```
Comment (KevLehman) [inline app/api/server/v1/groups.js:503]: The same comments as in the other file 🙈

File: app/api/server/v1/channels.js (hunk 1 above)
Comment (matheusbsilva137) [reply app/api/server/v1/channels.js:597]: Yes, `name` and `username` shouldn't be regexp.

File: app/api/server/v1/channels.js (hunk 2 above)
Comment (matheusbsilva137) [reply app/api/server/v1/channels.js]: Ok, I'm now using `check(status, Match.Maybe([String])`. Tks 😄

File: app/api/server/v1/channels.js [hunk 4]
```diff
@@ -577,6 +578,11 @@ API.v1.addRoute('channels.members', { authRequired: true }, {
...
 		const { offset, count } = this.getPaginationItems();
 		const { sort = {} } = this.parseJsonQuery();
+		const { status, username, name } = this.queryParams;
+
+		check(status, Match.Maybe([String]));
+		check(username, Match.Maybe(String));
+		check(name, Match.Maybe(String));
 
 		const subscriptions = Subscriptions.findByRoomId(findResult._id, {
 			fields: { 'u._id': 1 },
```
Comment (KevLehman) [inline app/api/server/v1/channels.js:588]: Probably not related to your PR, but, why is the offset/count passed to the subscriptions and not to the user's list? (which is the returned value) 👀

**Review by KevLehman — COMMENTED** [review]
The only thing that I think this is missing is support for search (by username, name, etc). But the basic support for the query params looks good :)

File: app/api/server/v1/channels.js (hunk 4 above)
Comment (sampaiodiego) [reply app/api/server/v1/channels.js]: right @KevLehman .. I have changed it now though =)

**Review by sampaiodiego — APPROVED** [review]

**Pull request merged**